main/src/ca/uwaterloo/flix/language/phase/Regions.scala

```diff
@@ -35,14 +35,11 @@ import scala.collection.immutable.SortedSet
 object Regions {
 
   def run(root: Root)(implicit flix: Flix): Validation[Unit, CompilationMessage] = flix.phase("Regions") {
-    val errs = ParOps.parMap(root.defs)(kv => visitDef(kv._2)).flatten
+    val recoverable = ParOps.parMap(root.defs)(kv => visitDef(kv._2)).flatten.filter(_.isInstanceOf[Recoverable])
```

Comment:
If you change the return type of visitDef to the specific error, I think everything works out (?) (i.e. without any instance of casts)?

Reply:
Nope still requires type casts